x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/search_source_expression.tsx

```diff
@@ -0,0 +1,221 @@
+  EuiEmptyPrompt,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Filter, ISearchSource } from '../../../../../../../src/plugins/data/common';
+import { EsQueryAlertParams, SearchType } from '../types';
+import {
+  ForLastExpression,
+  RuleTypeParamsExpressionProps,
+  ThresholdExpression,
+  ValueExpression,
+} from '../../../../../triggers_actions_ui/public';
+import { DEFAULT_VALUES } from '../constants';
+import { ReadOnlyFilterItems } from './read_only_filter_items';
+
+export const SearchSourceExpression = ({
+  ruleParams,
+  setRuleParams,
+  setRuleProperty,
+  data,
+  errors,
+}: RuleTypeParamsExpressionProps<EsQueryAlertParams<SearchType.searchSource>>) => {
+  const {
+    searchConfiguration,
+    thresholdComparator,
+    threshold,
+    timeWindowSize,
+    timeWindowUnit,
+    size,
+  } = ruleParams;
+  const [usedSearchSource, setUsedSearchSource] = useState<ISearchSource | undefined>();
+  const [paramsError, setParamsError] = useState<Error | undefined>();
+
+  const [currentAlertParams, setCurrentAlertParams] = useState<
+    EsQueryAlertParams<SearchType.searchSource>
+  >({
+    searchConfiguration,
+    searchType: SearchType.searchSource,
+    timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+    timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+    threshold: threshold ?? DEFAULT_VALUES.THRESHOLD,
+    thresholdComparator: thresholdComparator ?? DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+    size: size ?? DEFAULT_VALUES.SIZE,
+  });
+
+  const setParam = useCallback(
+    (paramField: string, paramValue: unknown) => {
+      setCurrentAlertParams((currentParams) => ({
+        ...currentParams,
+        [paramField]: paramValue,
+      }));
+      setRuleParams(paramField, paramValue);
+    },
+    [setRuleParams]
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => setRuleProperty('params', currentAlertParams), []);
+
+  useEffect(() => {
+    async function initSearchSource() {
+      try {
+        const loadedSearchSource = await data.search.searchSource.create(searchConfiguration);
+        setUsedSearchSource(loadedSearchSource);
+      } catch (error) {
+        setParamsError(error);
+      }
+    }
+    if (searchConfiguration) {
+      initSearchSource();
+    }
+  }, [data.search.searchSource, searchConfiguration]);
+
+  if (paramsError) {
+    return (
+      <>
+        <EuiCallOut color="danger" iconType="alert">
+          <p>{paramsError.message}</p>
+        </EuiCallOut>
+        <EuiSpacer size="s" />
+      </>
+    );
+  }
+
+  if (!usedSearchSource) {
+    return <EuiEmptyPrompt title={<EuiLoadingSpinner size="xl" />} />;
+  }
+
+  const dataView = usedSearchSource.getField('index')!;
+  const query = usedSearchSource.getField('query')!;
+  const filters = (usedSearchSource.getField('filter') as Filter[]).filter(
+    ({ meta }) => !meta.disabled
+  );
+  const indexPatterns = [dataView];
```

Comment:
why not name it `dataViews` from the start, cause this is a direction in which we're going anyway?